array.c

```diff
@@ -3879,23 +3893,25 @@ rb_ary_sort_by_bang(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.collect {|item| block}    -> new_ary
- *     ary.map     {|item| block}    -> new_ary
- *     ary.collect                   -> Enumerator
- *     ary.map                       -> Enumerator
- *
- *  Invokes the given block once for each element of +self+.
+ *    array.map {|element| ... } -> new_array
+ *    array.map -> new_enumerator
+ *    array.collect {|element| ... } -> new_array
+ *    array.collect -> new_enumerator
  *
- *  Creates a new array containing the values returned by the block.
+ *  Array#map is an alias for Array#collect.
  *
- *  See also Enumerable#collect.
+ *  Calls the block, if given, with each element of +self+;
+ *  returns a new \Array whose elements are the return values from the block:
+ *    a = [:foo, 'bar', 2]
+ *    a1 = a.collect {|element| element.class }
```

Comment:
That's not what's in the current doc for Array.  Often no examples at all;  no examples of both aliases.  Anyone else have an opinion?  @drbrain?

Reply:
I mean like this:
https://github.com/ruby/ruby/blob/5dfe95d10ec1dfeb4e114911367d37214c3b3f07/array.c#L3895-L3898

Here `collect` code has an example (line 3897) of `map`, since they are aliased, the documentation is the same for both.

Reply:
I know what you mean.  Almost none of the aliased methods in current doc for Array have both.  So I'd like to hear other opinions about adding those.

Reply:
A benefit of having examples of both aliases is that it gives both equal weight and takes away an argument like "well, the documentation only shows a `collect` example so `map` isn't recommended". In most of the code I read people prefer `map` over `collect` so I doubt anyone is coming to the documentation to try to make this poor argument.

I can't commit myself to a strong opinion on whether both aliases should appear in examples or not.  It'd be nice to have, but if we already doing a sufficiently good job of saying "both aliases exist together and are equivalent" it is redundant.

I would like to leave this up to @BurdetteLamar to decide.

Reply:
This is the 9th method I've done that has an alias.  Let's leave this alone here, and I'll add examples to all nine in my next PR.